algorithms/registration-toolbox/include/registration-toolbox/alignment/pcl-alignment.h

```diff
@@ -45,42 +55,137 @@ RegistrationResult PclAlignment<T_alignment, T_point>::registerCloudImpl(
   CHECK_NOTNULL(target);
   CHECK_NOTNULL(source);
 
-  PclPointCloudPtr<T_point> target_ds(new pcl::PointCloud<T_point>);
-  PclPointCloudPtr<T_point> source_ds(new pcl::PointCloud<T_point>);
+  PclPointCloudPtr<T_point> target_aligner_input;
+  PclPointCloudPtr<T_point> source_aligner_input;
 
-  voxel_grid_.setInputCloud(target);
-  voxel_grid_.filter(*target_ds);
-  voxel_grid_.setInputCloud(source);
-  voxel_grid_.filter(*source_ds);
+  if (FLAGS_regbox_pcl_use_downsampling) {
+    target_aligner_input =
+        PclPointCloudPtr<T_point>(new pcl::PointCloud<T_point>);
+    source_aligner_input =
+        PclPointCloudPtr<T_point>(new pcl::PointCloud<T_point>);
+    downSamplePointCloud(target, target_aligner_input);
+    downSamplePointCloud(source, source_aligner_input);
+  } else {
+    target_aligner_input = target;
+    source_aligner_input = source;
+  }
 
   bool is_converged = true;
   PclPointCloudPtr<T_point> transformed(new pcl::PointCloud<T_point>);
-  aligner_.setMaximumIterations(300);
-  aligner_.setRANSACIterations(1);
-  aligner_.setTransformationEpsilon(0.00001);
-  aligner_.setEuclideanFitnessEpsilon(1);
-  aligner_.setRANSACOutlierRejectionThreshold(0.05);
-  aligner_.setMaxCorrespondenceDistance(0.05);
-  aligner_.setUseReciprocalCorrespondences(false);
+  aligner_.setMaximumIterations(FLAGS_regbox_pcl_max_iterations);
 
   try {
     const Eigen::Matrix4f prior =
         prior_T_target_source.getTransformationMatrix().cast<float>();
-    aligner_.setInputTarget(target_ds);
-    aligner_.setInputSource(source_ds);
+    aligner_.setInputTarget(target_aligner_input);
+    aligner_.setInputSource(source_aligner_input);
     aligner_.align(*transformed, prior);
   } catch (std::exception& e) {
     LOG(ERROR) << "PCL registration failed to align the point clouds.";
     VLOG(3) << "PCL registration error: " << e.what();
     is_converged = false;  // just a procaution.
   }
+
+  const double fitness_score = aligner_.getFitnessScore(
+      FLAGS_regbox_pcl_fitness_max_considered_distance_m);
+  is_converged &= fitness_score <= FLAGS_regbox_pcl_max_fitness_score_m;
   const Eigen::Matrix4f T = aligner_.getFinalTransformation();
   const Eigen::MatrixXd cov = Eigen::MatrixXd::Identity(6, 6) * 1;
   return RegistrationResult(
       transformed, cov, this->convertEigenToKindr(T.cast<double>()),
       aligner_.hasConverged() & is_converged);
 }
 
+template <typename T_alignment, typename T_point>
+void PclAlignment<T_alignment, T_point>::downSamplePointCloud(
```

Comment:
this is pretty much the same as in https://github.com/ethz-asl/darpa_subt_base_station/blob/devel/alice/pointcloud_reporting_node/src/pointcloud-reporter.cc#L149